From 7d34ce461c129744713755b82f49c72591d1a301 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 28 Oct 2015 21:54:47 +0800
Subject: [PATCH] kgdboc, zynq: rx polling hook for the xilinx_uartps driver

The RX polling hook allows the debugger to hook character input so
as to allow entry to the kernel debugger with a control-c as an
example.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/tty/serial/xilinx_uartps.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/xilinx_uartps.c b/drivers/tty/serial/xilinx_uartps.c
index 3ddbac7..b46afae 100644
--- a/drivers/tty/serial/xilinx_uartps.c
+++ b/drivers/tty/serial/xilinx_uartps.c
@@ -234,6 +234,10 @@ static irqreturn_t cdns_uart_isr(int irq, void *dev_id)
 					continue;
 			}
 
+#ifdef CONFIG_CONSOLE_POLL
+			if (port->poll_rx_cb && port->poll_rx_cb((unsigned char)data))
+				continue;
+#endif
 #ifdef SUPPORT_SYSRQ
 			/*
 			 * uart_handle_sysrq_char() doesn't work if
-- 
2.0.2

